common/src/main/java/org/opensearch/ml/common/agent/MLToolSpec.java

```diff
@@ -56,6 +70,9 @@ public MLToolSpec(StreamInput input) throws IOException {
             parameters = input.readMap(StreamInput::readString, StreamInput::readOptionalString);
         }
         includeOutputInAgentResponse = input.readBoolean();
+        if (input.getVersion().onOrAfter(MINIMAL_SUPPORTED_VERSION_FOR_TOOL_CONFIG) && input.readBoolean()) {
+            configMap = input.readMap(StreamInput::readOptionalString, StreamInput::readOptionalString);
```

Comment:
This seems odd here, it looks like we're allowing null keys and null values in the map which I doubt is the case (at least the keys shouldn't be optional, and I'm not sure of the benefit of null values vs. just omitting the key).  Same comment in the `writeTo()` below.

Reply:
For the key, changed from `readOptionalString/writeOptionalString` to `readString/writeString`.